src/react/components/pages/homepage/homePage.tsx

```diff
@@ -122,12 +144,21 @@ export default class HomePage extends React.Component<IHomepageProps> {
         let projectJson: IProject;
 
         try {
-            projectJson = JSON.parse(project);
+            projectJson = JSON.parse(project.content);
         } catch (error) {
             throw new AppError(ErrorCode.ProjectInvalidJson, "Error parsing JSON");
         }
 
-        await this.loadSelectedProject(projectJson);
+        // need a better check to tell if its v1
+        if (projectJson.name == null) {
+            try {
+                await this.importConfirm.current.open(project);
```

Comment:
I have an idea to make this easier... I can return a promise from the `.open(...)` call that returns a value indicating whether or not the user confirmed the dialog.

Reply:
Do you mean in order to check if it is a v1 file? If this pops up every time the user tries to open a file, wouldn't that be bad for the user, especially if they have mostly v2 projects?

Reply:
This was just a note for myself to improve the experience for opening a modal that could return a promise vs. today it just returns void.